### ISSUE TITLE
[PSaaS Bug]: Log4j security flag raised by IT Security

### ISSUE THREAD

**Issue by BadgerOnABike** [PSaaS-Developers/Project_issues#2]
### Contact Details

brett.moore@nrcan-rncan

### What happened?

Our IT has detected Log4j within the prometheus manager and flagged it for patch/removal.

Please identify a path forward that we may remove Log4j and avoid these issues with all users.

While I have selected a version, it affects all versions.

### Version

(Dev) 2021.xx.xx

### What version of Windows are you seeing the problem on?

Windows 10 64-bit

### Relevant log output

```shell
C:\Program Files\Prometheus\manager\lib\log4j-1.2.17.jar

Log4j 2 is a Java-based logging library that is widely used in business system development, included in various open-source libraries and directly embedded in many software applications.

These vulnerabilities are tracked as CVE-2021-44832, CVE-2021-45105, CVE-2021-45046 and CVE-2021-44228, and referred to as “Log4Shell,” affecting Java-based applications that use Log4j 2 versions 2.0-beta7 through 2.17.0.

You must upgrade to version 2.17.1 (Java 8), 2.12.4 (Java 7), 2.3.2 (Java 6) or remove Log4j or the application that contains it within the next 5 business days.
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by spydmobile** [comment]
@tredpath Will we be able to address the log4j issues in the currently awaited build?

**Comment by tredpath** [comment]
It depends on how we can fix this. If just upgrading to the latest version where the issue has been resolved is allowed then this will be a mostly trivial change within our own code. If they are going to flag and remove any version of log4j then alternative logging packages would need to be investigated before the time required to update could be estimated.

The file referenced by Brett is also a dependency of a dependency so we will have to investigate where it came from and whether the third party has updated, can be dropped, or can have the dependency excluded without breaking it. Oddly, of the three log4j libraries bundled with Manager that is the only one not affected by this vulnerability.

The vulnerability is also not platform specific, it allows remote code execution on all platforms under the correct circumstances.

**Comment by BadgerOnABike** [comment]
It does appear that we may upgrade based on this statement:

> You must upgrade to version 2.17.1 (Java 8), 2.12.4 (Java 7), 2.3.2 (Java 6) or remove Log4j or the application that contains it within the next 5 business days.

**Comment by RobBryce** [comment]
Can you clarify on whether an upgrade will work, or if we need to find an alternative?

**Comment by BadgerOnABike** [comment]
Seems like we can upgrade per their statements in my initial posting.

**Comment by BadgerOnABike** [comment]
Is anything progressing with this or are we waiting on funding?

**Comment by RobBryce** [comment]
I've been given stop orders until funding is secured.

**Comment by spydmobile** [comment]
Clearly V2 is vulnerable, NOT V1, we are using V1, so while at EOL, not directly at risk, Tose instructions are clearly for those with the listed vulnerable versions, which we are NOT using, so I suggest we make a plan, not an emergency action, to migrate PAST the vulnerable versions, In the mean time, @BadgerOnABike you might try and explain how this vulnerability works and explain to IT that PSaaS And Log4J V1.2 is NOT vulnerable and that our upgrade path in next versions will take us PAST the vulnerable version to a safe one, but IMO we are not vulnerable at this time. Folks Feel free to set me straight if I have missed something.

**Comment by spydmobile** [comment]
Ok, I dont know where 1.2 came from, but from what I have seen, we might be using both, or just the vulnerable 2.x, @RobBryce I think you need to investigate the facts of the matter and report back here please. 
Which versions of this vulnerable library are being used in what parts?

**Comment by spydmobile** [comment]
SIDENOTE: @BadgerOnABike is this related to the uninstaller not taking old versions out?

**Comment by tredpath** [comment]
We are using the vulnerable versions, it is the core of our logging across all Java based Prometheus projects (Manager, Builder, REDapp, plus their underlying libraries used in Prometheus/PSaaS itself for loading weather). Updating those should be relatively easy. Another third party library is the 1.2 version which is not vulnerable but is EOL and appears to be getting flagged by IT when they do a simple version less that scan.

**Comment by BadgerOnABike** [comment]
Fortunately, the majority of folks aren't running PSaaS so our list of affected individuals is small.

**Comment by RobBryce** [comment]
> [...] @RobBryce I think you need to investigate the facts of the matter and report back here please. Which versions of this vulnerable library are being used in what parts?

Neal was clear on his stop orders on all work items until funding is figured out.

**Comment by spydmobile** [comment]
> Neal was clear on his stop orders on all work items until funding is figured out.
Yup understood @RobBryce, I am not suggesting you go to work, I am suggesting a future course of action and looking for your consensus ;-)

**Comment by BadgerOnABike** [comment]
Considering this is a major version update. 

Do you have an idea of how long this may take to resolve and what it would cost to achieve?

**Comment by spydmobile** [comment]
this is now superseded by #33
